geomesa-filter/src/test/scala/org/locationtech/geomesa/filter/FilterHelperTest.scala

```diff
@@ -41,6 +44,31 @@ class FilterHelperTest extends Specification {
 
   "FilterHelper" should {
 
+    "optimize the inArray function" >> {
+      val filter = ff.function("inArray", ff.literal("foo"), ff.literal(Array("foo", "bar")))
```

Comment:
doesn't seem like this filter is checked at all? curious if the literal array will work ok or if you need the array list

Reply:
https://github.com/geoserver/geoserver/blob/2.17.3/src/extension/querylayer/src/main/java/org/geoserver/filter/function/QueryFunction.java#L124-L151

An ArrayList is going to be generated.  Of course, this doesn't seem like a public API / fixed thing, so yeah....